src/components/media-player/dialog-media-player-browse.ts

```diff
@@ -223,30 +223,45 @@ class DialogMediaPlayerBrowse extends LitElement {
   static get styles(): CSSResultGroup {
     return [
       haStyleDialog,
+      haStyleDialogFixedTop,
       css`
         ha-dialog {
           --dialog-z-index: 9;
           --dialog-content-padding: 0;
         }
 
         ha-media-player-browse {
-          --media-browser-max-height: calc(100vh - 65px);
+          --media-browser-max-height: calc(
+            100vh -
+              65px - var(--safe-area-inset-top, var(--ha-space-0)) - var(
+                --safe-area-inset-bottom,
+                var(--ha-space-0)
+              )
+          );
```

Comment:
This can be simplified, I think
```suggestion
          --media-browser-max-height: calc(100vh - 65px - var(--safe-area-inset-y));
```